dev-docs/howtoguides/how_to_setup_windows_machine_for_wsl_testing.md

```diff
@@ -0,0 +1,181 @@
+integration tests.
+
+After that, select the `Windows 11` image and mark the `Security Type` as *Standard*.
+![](windows-security-type.png)
+
+The alternative "Trusted Launch" [doesn't support Nested Virtualization](https://learn.microsoft.com/en-us/azure/virtual-machines/trusted-launch#unsupported-features),
+required to run WSL.
+
+Set a username and password for the VM. Please set the username as *ubuntu*. 
+We will use this account setting to manually log into the machine to run some commands there directly.
+![](windows-ssh-setup.png)
+
+Finally, remember to also select the SSH inbound port for the machine
+![](windows-setup-port.png)
+
+
+## Configure Bastion to log into the machine
+
+To log into the machine and access its GUI, you need to set up Bastion.
+After launching the machine, go for the *Connect via Bastion* option:
+
+![](windows-bastion.png)
+
+![](windows-deploy-bastion.png)
+
+After deploying the Bastion, you just need to present your username and password to log into the
+machine.
+
+## Configuring Hyper-V
+
+After logging into the machine through Bastion, you need to configure Hyper-V by following
+this [tutorial](https://learn.microsoft.com/en-us/azure/lab-services/how-to-enable-nested-virtualization-template-vm-using-script?tabs=powershell)
+
+## Installing winget
+
+Each WSL Ubuntu distro has a dedicated installer in the Microsoft store. Therefore, before launching
+any WSL instance, we need to install the installers first. To achieve that, we need to configure
+**winget**.
+
+To install it on the machine, follow this [tutorial](https://winget.pro/winget-install-powershell/)
+
+```
+if you see the following error during installation, restart the machine first:
+
+Add-AppxPackage : Deployment failed with HRESULT: 0x80073D02, The package could not be installed
+because resources it modifies are currently in use. 
+```
+
+## Installing WSL
+
+To install WSL, just run the following command, as an administrator, on powershell:
+
+```console
+$ wsl --install
+```
+
+After that, we need to update WSL to the pre-release version by running:
+
+```console
+$ wsl --update --pre-release
+```
+
+
+```{note}
+We are installing the pre-release version because it allow us to directly run wsl commands
+through SSH. Once that fix is officially release, we will no longer need to install the
+pre-release version.
+```
+
+Now, install a Bionic instance to test the whole WSL set up is working as expected:
+
+```console
+$ winget install --name "Ubuntu 18.04 LTS" --accept-source-agreements --accept-package-agreements --silent
+```
+
+Now, create the instance with:
+
+```console
+$ & ubuntu1804.exe install --root
+```
+
+After that, just test to see if we can run commands on the WSL instance:
+
+```console
+$ wsl -d Ubuntu-18.04 --exec lsb_release -a
+```
+
+## Installing OpenSSH server
+
+To install OpenSSH server, follow these steps:
+
+1) Open *Settings*
```

Comment:
I think I might have overlooked something, but what application are we using here?